src/components/FileReader.vue

```diff
@@ -0,0 +1,68 @@
+<script lang="ts" setup>
+import { ref, reactive, computed } from 'vue'
+import { useImportStore } from '@/stores/import'
+import type MatlabFile from "@/interfaces/IExpectedFileContent"
+
+const docFile: any = ref(null)
+const importFile: any = useImportStore()
+
+const file: any = reactive({})
+const content: any = ref("")
+const loading: any = ref(false)
+
+function readMatlabTxtFile(): void {
+    file.value = docFile.value.files[0];
+    const reader = new FileReader();
+    if (file.value.name.includes(".txt")) {
+        reader.onload = (res) => {
+            content.value = res.target!.result;
+        };
+        reader.readAsText(file.value);
+    }
+}
+
+function parseMatLabTxtData(): void {
+    loading.value = true
+    const pairs = content.value.split("\n")
+    const parsedContent: MatlabFile[] = []
+    for (let i = 0; i < pairs.length; i++) {
+        const pair = pairs[i];
+        const indexEndOfLeftValue = pair.indexOf("\t")
+        const indexEndOfRightValue = pair.indexOf("\r")
+        const parsedValuesObject = {
+            x: Number(pair.substring(0, indexEndOfLeftValue)),
+            y: Number(pair.substring((indexEndOfLeftValue + 1), indexEndOfRightValue))
+        }
+        parsedContent.push(parsedValuesObject)
+    }
+    importFile.fileContent = parsedContent
+    setTimeout(() => (loading.value = false), 500)
```

Comment:
Why the timeout? Just set loading to false when you're done parsing the file?

Reply:
This has something to do with how the graphs are loaded. Something strange happens in `Graph.vue` line 66 related to the library and this timeout simply interacts with that to make sure the loading icon displays properly for user. Another US should pick this up.